protected/humhub/modules/dashboard/stream/filters/DashboardMemberStreamFilter.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace humhub\modules\dashboard\stream\filters;
+
+use humhub\modules\friendship\models\Friendship;
+use Yii;
+use humhub\modules\content\models\Content;
+use humhub\modules\dashboard\Module;
+use humhub\modules\space\models\Membership;
+use humhub\modules\stream\models\filters\StreamQueryFilter;
+use humhub\modules\space\models\Space;
+use humhub\modules\user\models\User;
+
+/**
+ * Stream filter handling dashboard content stream visibility for members of the network.
+ *
+ * @since 1.8
+ */
+class DashboardMemberStreamFilter extends StreamQueryFilter
+{
+    /**
+     * @var User
+     */
+    public $user;
+
+    /**
+     * @inheritDoc
+     */
+    public function apply()
+    {
+        $this->joinWithSubscribedContainers();
+        $this->filterSubscribedContainer();
+        $this->filterContentVisibility();
+        $this->query->addParams([
+            ':userId' => $this->user->id,
+            ':spaceMembershipStatus' => Membership::STATUS_MEMBER,
+            ':spaceEnabledStatus' => Space::STATUS_ENABLED,
+            ':userEnabledStatus' => User::STATUS_ENABLED,
+            ':userModel' => User::class,
+            ':spaceModel' => Space::class,
+            ':visibilityPrivate' => Content::VISIBILITY_PRIVATE,
+            ':visibilityPublic' => Content::VISIBILITY_PUBLIC,
+            ':userContentContainerId' => $this->user->contentcontainer_id
+        ]);
+    }
+
+    /**
+     * Adds joins for container subscription checks
+     */
+    private function joinWithSubscribedContainers()
+    {
+        // Join with enabled space containers
```

Comment:
Didn't we plan to use the cached Live Legitimation IDs here?

Reply:
I've implemented a legitimation based dashboard filter for testing, but it was not scalable and overall not much faster as the new solution. The query of this PR is ca 30% faster as the old one.

Reply:
Here is the dummy legitimation filter which is ~50% slower than the join based one in case of many container subscriptions:

```
 private function filterContainerIds($friendshipEnabled)
    {
        /* @var $dashboardModule \humhub\modules\live\Module */
        $dashboardModule = Yii::$app->getModule('live');

        if(static::$legitimation === null) {
            static::$legitimation = $dashboardModule->getLegitimateContentContainerIds($this->user);
        }

       // $isFollowAllProfileActive = $this->isFollowAllProfilesActive();

        $containerIdCondition = ['OR'];

        $containerIdCondition[] = ['and',
            ['content.visibility' => Content::VISIBILITY_PUBLIC],
            ['IN', 'content.contentcontainer_id', static::$legitimation[Content::VISIBILITY_PUBLIC]]
        ];

        $containerIdCondition[] = ['and',
            ['content.visibility' => Content::VISIBILITY_PRIVATE],
            ['IN', 'content.contentcontainer_id', static::$legitimation[Content::VISIBILITY_PRIVATE]]
        ];

        $containerIdCondition[] = ['contentcontainer.id' => $this->user->contentcontainer_id];
        $containerIdCondition[] = 'contentcontainer.id IS NULL';
        $containerIdCondition[] = ['content.created_by' => $this->user->id];

       /* if ($isFollowAllProfileActive) {
            // INCLUDE ALL USER PROFILES
            // TODO: get rid of user follower related queries in this case
            $containerIdCondition[] =  ['IN', 'contentcontainer.id', (new Query())->select(["allusers.contentcontainer_id"])->from('user allusers')];
        }*/

        $this->query->andFilterWhere($containerIdCondition);
    }
```